src/create-jobs.js

```diff
@@ -244,7 +247,13 @@ const excludeJob = function (job, exclusion) {
                 return;
             }
 
-            if (job[jobKey] === exclusion[excludeKey]) {
+            let jobValue = job[jobKey],
+                exclusionValue = exclusion[excludeKey];
+
+            if (jobValue === exclusionValue) {
+                matches += 1;
+            } else if (excludeKey === 'name' && job.deprecatedName !== '' && exclusionValue === job.deprecatedName) {
```

Comment:
This is actually the BC compat layer for the issue mentioned in https://github.com/laminas/laminas-ci-matrix-action/issues/76#issuecomment-1057746570